packages/@aws-cdk/aws-ecr/test/repository.test.ts

```diff
@@ -363,14 +362,29 @@ describe('repository', () => {
 
     // WHEN
     repo.addToResourcePolicy(new iam.PolicyStatement({
-      resources: ['*'],
       actions: ['ecr:*'],
     }));
 
     // THEN
     expect(() => app.synth()).toThrow(/A PolicyStatement used in a resource-based policy must specify at least one IAM principal/);
   });
 
+  test('warns if repository policy has resources', () => {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'my-stack');
+    const repo = new ecr.Repository(stack, 'Repo');
+
+    // WHEN
+    repo.addToResourcePolicy(new iam.PolicyStatement({
+      resources: ['*'],
+      actions: ['ecr:*'],
```

Comment:
I think you have to specify a principal here.

Reply:
@daschaa thank you! I'll try that!